src/Language/DifferentialDatalog/Type.hs

```diff
@@ -28,9 +28,10 @@ module Language.DifferentialDatalog.Type(
     typeUserTypes,
     typeIsPolymorphic,
     unifyTypes,
-    exprType, exprType', exprType'', exprTypeMaybe, exprNodeType,
---    exprTypes,
-    exprTraverseTypeM,
+    exprType,
+    exprType',
```

Comment:
I am a bit worried that you need to export primed names; this means either that the names are wrong, or that you are not hiding enough.

Reply:
I guess you're right, but these functions (`typ`, `typ'`, `typ''` and `exprType`, `exprType'`, `exprType'''`) are ubiquitous and I like using short names for them. Otherwise they'd have to be called, e.g., `typ`, `typCanonical`, `typ????` (not even sure what the name should be)